### ISSUE TITLE
Rewrite Builder to use `mix compile`

### ISSUE THREAD

**Issue by gw4e** [KronicDeth/intellij-elixir#918]
Hi,
With a file with compilation error , running a full build of the project does not lead to display compilation error messages in the Message tab.
<img width="1379" alt="intellij_eixir" src="https://user-images.githubusercontent.com/28825472/33184734-65d1e82a-d07e-11e7-9404-df014bf9fffb.png">
Regards.

**Comment by KronicDeth** [comment]
yeah, the compile through "Build" menu was a really earlier feature of the plugin before I really knew what I was doing (and most of it came in from other contributors PRs too, so I didn't have as strong an understanding of it as I would not).  It really needs to be re-written to use the current `mix` system and `mix compile`.

**Comment by KronicDeth** [comment]
@gw4e I usually just have the compiles happen indirectly from the Run Configurations for Mix ExUnit or Mix with `phoenix.server`, etc.  It make work as a work-around for you.

You can also setup `mix compile` in a Mix Run Configuration by having `compile` as the only argument as a more precise work-around for now.

**Comment by sh41** [comment]
This is done — the builder uses `mix compile` by default.

The Mix build path is selected by the `useMixCompiler` compiler option, and what resolved this issue was making it the default, in commit f2a263c23b0cb616fbd08fc7c3d8b155b0b12f45 ("Use mix compiler and attach docs by default in compiler"). That shipped in v7.4.0.

Building now goes through Mix rather than the plugin's own `elixirc` invocation, so compiled `.beam` files land in Mix's `_build/<env>/lib/<app>/ebin/` layout rather than a plugin-specific output directory. The build tests assert that layout, so it's pinned rather than incidental.

Closed as part of an AI-assisted review of the backlog. A human checked this one before closing it, but it might still be a mistake. If you think the reason above is wrong, please reopen the issue and tell us why.
